src/Microsoft.NetCore.Analyzers/Core/Security/DoNotUseInsecureDeserializerWithoutBinderBase.cs

```diff
@@ -0,0 +1,196 @@
+                                },
+                                OperationKind.MethodReference);
+
+                            operationBlockStartAnalysisContext.RegisterOperationBlockEndAction(
+                                (OperationBlockAnalysisContext operationBlockAnalysisContext) =>
+                                {
+                                    if (!rootOperationsNeedingAnalysis.Any())
+                                    {
+                                        return;
+                                    }
+
+                                    // Only instantiated if there are any results to report.
+                                    Dictionary<OperationMethodKey, PropertySetAbstractValue> allResults = null;
+
+                                    foreach (IOperation rootOperation in rootOperationsNeedingAnalysis)
+                                    {
+                                        ImmutableDictionary<OperationMethodKey, PropertySetAbstractValue> dfaResult =
+                                            PropertySetAnalysis.GetOrComputeHazardousParameterUsages(
+                                                rootOperation.GetEnclosingControlFlowGraph(),
+                                                operationBlockAnalysisContext.Compilation,
+                                                operationBlockAnalysisContext.OwningSymbol,
+                                                this.DeserializerTypeMetadataName,
+                                                true /* isNewInstanceFlagged */,
+                                                this.SerializationBinderPropertyMetadataName,
+                                                true /* isNullPropertyFlagged */,
+                                                cachedDeserializationMethodNames);
+                                        if (!dfaResult.Any())
+                                        {
+                                            continue;
+                                        }
+
+                                        if (allResults == null)
+                                        {
+                                            allResults = new Dictionary<OperationMethodKey, PropertySetAbstractValue>();
+                                        }
+
+                                        foreach (KeyValuePair<OperationMethodKey, PropertySetAbstractValue> kvp in dfaResult)
+                                        {
+                                            allResults.Add(kvp.Key, kvp.Value);
+                                        }
+                                    }
+
+                                    if (allResults == null)
+                                    {
+                                        return;
+                                    }
+
+                                    foreach (OperationMethodKey operationMethodKey in allResults.Keys.OrderBy(OperationMethodKey.Comparer))
```

Comment:
>OrderBy(OperationMethodKey.Comparer) [](start = 102, length = 36)

Why do you need the keys to be processed in a specific order? I believe you can completely avoid `OperationMethodKey` and all the comparers added in this PR.

Reply:
Ah, I presume you only want to flag the first hazardous usage. How about using `Dictionary<(Location location, IMethodSymbol method), PropertySetAbstractValue>` and then `allResults.Orderby(kvp => kvp.Key.location.SpanStart)`?

---
In reply to: [224088958](https://github.com/dotnet/roslyn-analyzers/pull/1838#discussion_r224088958) [](ancestors = 224088958)

Reply:
I wanted consistent ordering in the diagnostics reported.

Reply:
An analyzer cannot control that, compiler always sorts the reported diagnostic by file/location. You should report diagnostics without worrying about ordering.

Reply:
Oh nice, I'll remove all this then.  Maybe I'll need to do something else for unit tests not caring about ordering.  I think I had some failures.

Reply:
I believe order does not matter for unit tests verification, we basically do set equality.

---
In reply to: [224187522](https://github.com/dotnet/roslyn-analyzers/pull/1838#discussion_r224187522) [](ancestors = 224187522)